### PR TITLE
Feature/mico424 delete all services

### LINKED ISSUE #424: implement Delete all service versions endpoint
As a UI-Developer, I want to have an endpoint to delete all versions of a service.
The services should only be deleted, if they are not deployed and not required by an application.

### PR DESCRIPTION
**Pull request by Pascal157600**
- [ ] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Ensure to use auto format in **all** files
- [ ] Tests created for changes or:
  - [ ] Test cases are missing - opened issue for that:
  - [ ] Reason for missing tests:
- [ ] Screenshots added (for UI changes)
- [ ] Change in CHANGELOG.md described

---

## Resolving Issue/ Feature

Resolves #424

### CODE REVIEW

File: mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java [hunk 1]
```diff
@@ -124,21 +124,34 @@
...
         serviceRepository.deleteServiceByShortNameAndVersion(shortName, version);
         return ResponseEntity.noContent().build();
     }
 
+    @DeleteMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}")
+    public ResponseEntity<Void> deleteServices(@PathVariable(PATH_VARIABLE_SHORT_NAME) String shortName) {
+        List<MicoService> micoServiceList = getServicesFromDatabase(shortName);
+        log.debug("Got following services from database: {}", micoServiceList);
+        micoServiceList.forEach(service -> {
+            //TODO: check for deployment
```
Comment (JanVetter) [inline mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java]: Issue created for this?

Comment (Pascal157600) [reply mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java]: aw yeahhh

File: mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java [hunk 2]
```diff
@@ -124,21 +124,34 @@
...
-                "Service '" + service.getShortName() + "' '" + service.getVersion() + "' has dependers, therefore it can't be deleted.");
+                    "Service '" + service.getShortName() + "' '" + service.getVersion() + "' has dependers, therefore it can't be deleted.");
         }
 
         serviceRepository.deleteServiceByShortNameAndVersion(shortName, version);
         return ResponseEntity.noContent().build();
     }
 
+    @DeleteMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}")
+    public ResponseEntity<Void> deleteServices(@PathVariable(PATH_VARIABLE_SHORT_NAME) String shortName) {
```
Comment (JanVetter) [inline mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java]: Maybe change function name to `deleteAllVersionsOfService` or something similar?

File: mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java [hunk 3]
```diff
@@ -365,6 +378,16 @@ private MicoService getServiceFromDatabase(String shortName, String version) thr
         return serviceOpt.get();
     }
 
+    private List<MicoService> getServicesFromDatabase(String shortName) throws ResponseStatusException {
```
Comment (JanVetter) [inline mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java]: Maybe change function name to `getVersionsOfServiceFromDatabase` or something similar?

File: mico-core/src/test/java/io/github/ust/mico/core/ServiceControllerTests.java [hunk 4]
```diff
@@ -450,51 +450,70 @@ public void deleteSpecificServiceDependee() throws Exception {
...
-            .contentType(MediaTypes.HAL_JSON_UTF8_VALUE))
-            .andDo(print());
+                .contentType(MediaTypes.HAL_JSON_UTF8_VALUE))
+                .andDo(print());
 
         resultDelete.andExpect(status().isCreated());
     }
 
+    @Test
+    public void deleteAllServices() throws Exception {
```
Comment (JanVetter) [inline mico-core/src/test/java/io/github/ust/mico/core/ServiceControllerTests.java]: 👍

**Pull request merged**